### PR TITLE
FIX(Readme) TESTING.md various docs updates

### PR DESCRIPTION
**Pull request by michalxo**
Signed-off-by: Michal Tóth <mtoth@redhat.com>

### Testing.md file updated for minor language corrections and typos etc 

- Documentation

### Description

Updated TESTING.md

### Checklist
- [ ] Update documentation

### CODE REVIEW

File: development-docs/TESTING.md [hunk 1]
```diff
@@ -90,18 +91,18 @@ Cluster Operator setup example:
     }
 ```
 
-##### Exercise
-In this phase you specify all the steps which you need to cover some functionality. 
-If you didn't create the Kafka cluster you should do so at the begging of test using the test method resources instance of `Resources` inherited from `BaseST`.
+### Exercise
+In this phase you specify all steps which you need to execute to cover some specific functionality.
+If you didn't create the Kafka cluster you should do so at the beginning of the test using a test method resources instance of `Resources` inherited from `BaseST`.
```
[the remark this answers is not in the document]
Comment (michalxo) [reply development-docs/TESTING.md]: fixed `BaseST` -> `AbstractST`

File: development-docs/TESTING.md [hunk 2]
```diff
@@ -90,19 +91,19 @@ Cluster Operator setup example:
     }
 ```
 
-##### Exercise
-In this phase you specify all the steps which you need to cover some functionality. 
-If you didn't create the Kafka cluster you should do so at the begging of test using the test method resources instance of `Resources` inherited from `BaseST`.
+### Exercise
+In this phase you specify all steps which you need to execute to cover some specific functionality.
+If you didn't create the Kafka cluster you should do so at the beginning of the test using a test method resources instance of `Resources` inherited from `AbstractST`.
```
Comment (Frawless) [inline development-docs/TESTING.md:96]: I don't thin this is valid now. We use two stacks and you can everywhere use static methods from "resources" classes (like `KafkaResources.class`) and `ResourceManager.class`will handle the rest. This approach in TESTING.md is quite old and I thing I just forgot to remove it.

File: development-docs/TESTING.md [hunk 3]
```diff
@@ -32,50 +32,51 @@ The next requirement is to have built the `systemtest` package dependencies, whi
...
 
-We create resources in Kubernetes cluster via classes in `resources` package, which allows you to deploy all components and, if needed, change them from their default configuration using a builder. 
-Currently, we have two stacks, which are stored in `ResourceManager` singleton instance — one for whole test class resources and one for test method resources.
-You can create resources anywhere you want, and our implementation will take care about put resource on top of one stack and delete them at the end of test method/class.
+We create resources in Kubernetes cluster via classes in `resources` package, which allows you to deploy all components and, if needed, change them from their default configuration using a builder.
+Currently, we have two stacks, which are stored in `ResourceManager` singleton instance — one for all test class resources and one for test method resources.
+You can create resources anywhere you want. Our resource lifecycle implementation will handle insertion of the resource on top of stack and deletion at the end of the test method/class.
```
Comment (im-konge) [inline development-docs/TESTING.md:74]: Just thinking - do we really want almost same sentence so many times? It's the third time until this line when I see this - `... resource lifecycle implementation will ... and deletion of resources at the end of tests`. 

Do we want this so many times in our docu? It's just idea.

File: development-docs/TESTING.md [hunk 4]
```diff
@@ -144,125 +145,124 @@ You need to use the `groups` system property in order to execute a group of syst
...
 
     EXTRA_ARGS="-Dfoo=bar" ./systemtest/scripts/run_tests.sh
-    
+
 ## Running single test class
 
-Use the `verify` build goal and provide a `-Dit.test=TestClassName[#testMethodName]` system property. 
+Use the `verify` build goal and provide `-Dit.test=TestClassName[#testMethodName]` system property.
 
     mvn verify -pl systemtest -Dit.test=KafkaST#testKafkaAndZookeeperScaleUpScaleDown
```
Comment (im-konge) [inline development-docs/TESTING.md:250]: Can you please change the `testKafkaAndZookeeperScaleUpScaleDown` to `testCustomAndUpdatedValues`? This test is no longer part of the `KafkaST` and if someone will try it, he will see some errors and maybe he will think that he is doing something wrong. Thanks :smile:

File: development-docs/TESTING.md (hunk 3 above)
Comment (michalxo) [reply development-docs/TESTING.md:74]: I think I can simplify it on lines  49-50. Let me have a look at it.

File: development-docs/TESTING.md (hunk 2 above)
Comment (michalxo) [reply development-docs/TESTING.md:96]: ok I am removing this line 96.

File: development-docs/TESTING.md (hunk 4 above)
Comment (michalxo) [reply development-docs/TESTING.md:250]: Done.

**Pull request merged**